File: mpi_adv_examples/win_create.f90
PROGRAM win_create
    USE mpi_f08
    IMPLICIT NONE
    INTEGER::a,b,myrank
    INTEGER(KIND=MPI_ADDRESS_KIND)::size,lb
    TYPE(MPI_Win)::win
    CALL MPI_Init
    CALL MPI_Comm_rank(MPI_COMM_WORLD,myrank)
    IF(myrank==0) a=1
    IF(myrank==1) a=10
    IF(myrank==2) a=100
    PRINT*,'myrnak:',myrank, 'a=',a
    size=STORAGE_SIZE(a)/8
    CALL MPI_Type_get_extent(MPI_INTEGER,lb,size)
    CALL MPI_Win_create(a,size,1,MPI_INFO_NULL,MPI_COMM_WORLD,win)
    CALL MPI_Win_free(win)
    CALL MPI_Finalize
END PROGRAM win_create
    

File: mpi_adv_examples/win_create.c
#include <stdio.h>
#include <mpi.h>
int main()
{
  int a,b,myrank;
  MPI_Win win;
  MPI_Init(NULL,NULL);
  MPI_Comm_rank(MPI_COMM_WORLD,&myrank);
  if(myrank==0) a=1;
  if(myrank==1) a=10;
  if(myrank==2) a=100;
  printf("myrank:%d   a=%d\n",myrank,a);
  MPI_Win_create(&a,sizeof(int),1,MPI_INFO_NULL,MPI_COMM_WORLD,&win);
  MPI_Win_free(&win);
  MPI_Finalize();
  return 0;
}
